clean/hammond_pd_cprr.py

```diff
@@ -163,6 +163,123 @@ def clean_action_09(df):
+        .str.replace('jeannine cruz', 'cruz jeannine', regex=False)\
+        .str.replace('vicki corkern', 'corkern vicki', regex=False)\
+        .str.replace('terry zaffuto', 'zaffuto terry', regex=False)\
+        .str.replace('james king', 'king james', regex=False)\
+        .str.replace('thomas miller', 'miller thomas', regex=False)\
+        .str.replace('derrick foster', 'foster derrick', regex=False)\
+        .str.replace(r'jo(seph|ey) drago', 'drago joseph', regex=True)
+
+    names = df.name.str.extract(r'(\w+),? (\w+)')
+    df.loc[:, 'last_name'] = names[0]
+    df.loc[:, 'first_name'] = names[1]
+    return df.drop(columns='name')
+
+
+def extract_complaint_type(df):
+    types = df.nature_of_incident.str.lower().str.strip()\
+        .str.extract(r'(citizen complaint)')
+    df.loc[:, 'complaint_classification'] = types[0]\
+        .str.replace('complaint', '', regex=False)
+    return df
+
+
+def clean_charges_08(df):
+    df.loc[:, 'charges'] = df.nature_of_incident.str.lower().str.strip()\
+        .str.replace('citizen complaint', '', regex=False)\
+        .str.replace(r'(\w+)[\,\;] (\w+)', r'\1/\2', regex=True)\
+        .str.replace(r'ia-? ?(threatening|alleged)? ', '', regex=True)\
+        .str.replace('&', 'and', regex=False)\
+        .str.replace('invest', 'investigation', regex=False)\
+        .str.replace('complaint ', '', regex=False)\
+        .str.replace(r'^conduct unbecoi?ming', 'conduct unbecoming an officer', regex=True)
+    return df.drop(columns='nature_of_incident')
+
+
+def extract_and_clean_action(df):
+    df.loc[:, 'suspension_hrs'] = df.suspension_hrs.str.lower().str.strip()\
+        .str.replace(r'(\d+) days? ?(susp)?', r'\1 day suspension', regex=True)
+
+    df.loc[:, 'letter_of_rep'] = df.letter_of_rep.str.lower().str.strip()\
+        .str.replace('x', 'letter of reprimand')
+
+    df.loc[:, 'action'] = df.no_action.str.lower().str.strip()\
+        .str.replace('x', '', regex=False)\
+        .str.replace('no action', '', regex=False)\
+        .str.replace('suspension - 1 day per gen order 107',
+                     '1 day suspension per gen order 107', regex=False)\
+        .str.replace(r'(\w+) - (\w+)', r'\1|\2', regex=True)
+
+    df.loc[:, 'action'] = df.action.fillna('') + '' + df.letter_of_rep.fillna('')
+    df.loc[:, 'action'] = df.action + '' + df.suspension_hrs.fillna('')
+
+    df.loc[:, 'action'] = df.action.str.replace('reprimand2', 'reprimand|2', regex=False)
+    return df.drop(columns=['no_action', 'letter_of_rep', 'suspension_hrs'])
+
+
+def extract_and_clean_dispositions_08(df):
+    df.loc[:, 'dismiss'] = df.dismiss.str.lower().str.strip()\
+        .str.replace('x', 'dismissed', regex=False)
+
+    df.loc[:, 'disposition'] = df.founded_un.str.lower().str.strip()\
+        .str.replace(r'^u$', 'unfounded', regex=True)\
+        .str.replace('x', 'founded', regex=False)
+
+    df.loc[:, 'disposition'] = df.disposition.fillna('') + '' + df.dismiss.fillna('')
```

Comment:
Perhaps you meant to put a single space instead of an empty string between these?

Reply:
Fixed this. No separation is needed because there is no overlap, instead dismissed is a missing disposition.